sync/src/main/java/org/bf2/sync/ManagedKafkaSync.java

```diff
@@ -152,15 +156,18 @@ boolean deleteAllowed(ManagedKafka local) {
         return true;
     }
 
-    public static boolean changed(ManagedKafka remoteManagedKafka, ManagedKafka existing) {
-        ManagedKafkaSpec remoteSpec = remoteManagedKafka.getSpec();
+    public boolean changed(ManagedKafka remote, ManagedKafka existing) {
+        ManagedKafkaSpec remoteSpec = remote.getSpec();
         if (!remoteSpec.isDeleted() && existing.getSpec().isDeleted()) {
             // TODO: seems like a problem / resurrection - should not happen
             log.warnf("Ignoring ManagedKafka %s that wants to come back to life", Cache.metaNamespaceKeyFunc(existing));
             return false;
         }
-
-        return !remoteSpec.equals(existing.getSpec()) || !Objects.equals(existing.getMetadata(), remoteManagedKafka.getMetadata());
+        if (secretManager.isMasterSecretChanged(remote, existing)) {
+            return true;
+        }
+        ManagedKafka remoteCopy = secretManager.removeSecretsFromManagedKafka(remote);
+        return !remoteCopy.getSpec().equals(existing.getSpec()) || !Objects.equals(existing.getMetadata(), remote.getMetadata());
```

Comment:
`remote.getMetadata()` maybe changed to `remoteCopy.getMetadata()` for consistency